server/models/registry.js

```diff
@@ -13,9 +13,19 @@ class Registry {
     return this.instances[id] ? id : null;
   }
 
+  restore(storedObject) {
+    let id = storedObject.id;
+    this.instances[id] = this.constructor.getInstanceClass().restore(storedObject);
+    return this.instances[id] ? id : null;
+  }
+
   get(id) {
     return this.instances[id];
   }
+
+  static getInstanceClass() {
```

Comment:
Hmm, this is an interesting approach, but we are using in this class two different strategies to customize subclasses of `Registry`. Here you are adding a static function that subclasses must implement; I previously opted to receive the custom behavior (in particular, the function to create new game instances) in the constructor. I can see a few advantages and disadvantages in each of the approaches, but I think I'm ok with either one - but we have to choose only one.

Can you please change this so that either:
* you pass the instance class as another constructor argument, which you can store and retrieve later;
* the `instanceFactory` parameter is converted into a static `_createGameInstance(id, params)` function that must be implemented by subclasses.